ethereum/retesteth/src/main/java/tech/pegasys/pantheon/ethereum/retesteth/RetestethContext.java

```diff
@@ -0,0 +1,286 @@
+import tech.pegasys.pantheon.ethereum.eth.manager.EthContext;
+import tech.pegasys.pantheon.ethereum.eth.manager.EthMessages;
+import tech.pegasys.pantheon.ethereum.eth.manager.EthPeers;
+import tech.pegasys.pantheon.ethereum.eth.manager.EthScheduler;
+import tech.pegasys.pantheon.ethereum.eth.sync.state.SyncState;
+import tech.pegasys.pantheon.ethereum.eth.transactions.PendingTransactions;
+import tech.pegasys.pantheon.ethereum.eth.transactions.TransactionPool;
+import tech.pegasys.pantheon.ethereum.eth.transactions.TransactionPoolConfiguration;
+import tech.pegasys.pantheon.ethereum.eth.transactions.TransactionPoolFactory;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.processor.BlockReplay;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.queries.BlockchainQueries;
+import tech.pegasys.pantheon.ethereum.mainnet.EthHashSolver;
+import tech.pegasys.pantheon.ethereum.mainnet.EthHasher;
+import tech.pegasys.pantheon.ethereum.mainnet.HeaderValidationMode;
+import tech.pegasys.pantheon.ethereum.mainnet.MainnetBlockHeaderFunctions;
+import tech.pegasys.pantheon.ethereum.mainnet.MainnetProtocolSchedule;
+import tech.pegasys.pantheon.ethereum.mainnet.ProtocolSchedule;
+import tech.pegasys.pantheon.ethereum.mainnet.ProtocolSpec;
+import tech.pegasys.pantheon.ethereum.storage.keyvalue.KeyValueStoragePrefixedKeyBlockchainStorage;
+import tech.pegasys.pantheon.ethereum.storage.keyvalue.WorldStateKeyValueStorage;
+import tech.pegasys.pantheon.ethereum.storage.keyvalue.WorldStatePreimageKeyValueStorage;
+import tech.pegasys.pantheon.ethereum.worldstate.WorldStateArchive;
+import tech.pegasys.pantheon.metrics.MetricsSystem;
+import tech.pegasys.pantheon.metrics.noop.NoOpMetricsSystem;
+import tech.pegasys.pantheon.services.kvstore.InMemoryKeyValueStorage;
+import tech.pegasys.pantheon.util.bytes.BytesValue;
+
+import java.util.Locale;
+import java.util.Optional;
+import java.util.concurrent.locks.ReentrantLock;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.base.Functions;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class RetestethContext {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final ReentrantLock contextLock = new ReentrantLock();
+  private Address coinbase;
+  private DefaultMutableBlockchain blockchain;
+  private ProtocolContext<Void> protocolContext;
+  private BlockchainQueries blockchainQueries;
+  private ProtocolSchedule<Void> protocolSchedule;
+  private HeaderValidationMode headerValidationMode;
+  private BlockReplay blockReplay;
+  private RetestethClock retesethClock;
+
+  private TransactionPool transactionPool;
+  private EthScheduler ethScheduler;
+  private EthHashSolver ethHashSolver;
+
+  public boolean resetContext(
+      final String genesisConfigString, final String sealEngine, final Optional<Long> clockTime) {
```

Comment:
My concern is how do we deal with new random values?  The testing team has already shown that they will introduce random undocumented values ("NoReward") and I expect at least one other arbitrary add ("ProgPow" or some variant of the name).  If we make an enum test evolution becomes a parsing error and we fail with an RPC exception instead of bad data.  Retesteh is already bad when it comes to reporting back data on the RPC failures (i.e. which test tripped it) but is much better about state failures.

So if PegaSys was controlling the code I'd say yes, but I don't feel we can trust the community code will behave well.

Reply:
👍Make sense to me as-is then